library/pkparse.c

```diff
@@ -667,7 +830,20 @@ int mbedtls_pk_parse_subpubkey(unsigned char **p, const unsigned char *end,
 #endif /* MBEDTLS_RSA_C */
 #if defined(MBEDTLS_ECP_LIGHT)
     if (pk_alg == MBEDTLS_PK_ECKEY_DH || pk_alg == MBEDTLS_PK_ECKEY) {
-        ret = pk_use_ecparams(&alg_params, &mbedtls_pk_ec(*pk)->grp);
+        switch (ec_grp_id) {
+#if defined(MBEDTLS_ECP_DP_CURVE25519_ENABLED)
+            case MBEDTLS_ECP_DP_CURVE25519:
+#endif
+#if defined(MBEDTLS_ECP_DP_CURVE448_ENABLED)
+            case MBEDTLS_ECP_DP_CURVE448:
+#endif
+#if defined(MBEDTLS_ECP_DP_CURVE25519_ENABLED) || defined(MBEDTLS_ECP_DP_CURVE448_ENABLED)
+                ret = pk_use_ecparams_curdle(&alg_params, ec_grp_id, &mbedtls_pk_ec(*pk)->grp);
+                break;
```

Comment:
Your formatting script is not able to understand this code. It wants to indent this twice to the left, which is incorrect. Same issue at line 1291.

Reply:
It seems that when all the `case` statements are guarded by a preprocessor directive, uncrustify doesn't realize that there is a list of `case` statements. The only workarounds I can think of:

* Leave at least `case MBEDTLS_ECP_DP_CURVE25519:` to be present when Curve448 is enabled but not Curve25519. The code size hit is acceptably tiny.
* Locally disable automatic indentation with `*INDENT-OFF*` / `*INDENT-ON*` comments. Seems a bit heavy-handed here.